ui/apps/platform/src/Containers/Violations/Details/Deployment/DeploymentTabWithReadAccessForDeployment.tsx

```diff
@@ -2,27 +2,22 @@ import React, { ReactElement } from 'react';
 import { Alert, Card, CardBody, CardTitle, Flex, FlexItem, Title } from '@patternfly/react-core';
 
 import useFetchDeployment from 'hooks/useFetchDeployment';
-import usePermissions from 'hooks/usePermissions';
-import { DeploymentAlert } from 'types/alert.proto';
 import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
 
+import { AlertDeployment } from 'types/alert.proto';
+
 import ContainerConfiguration from './ContainerConfiguration';
 import DeploymentOverview from './DeploymentOverview';
-import NetworkPoliciesCard from './NetworkPoliciesCard';
 import PortDescriptionList from './PortDescriptionList';
 import SecurityContext from './SecurityContext';
 
-export type DeploymentDetailsProps = {
-    alertDeployment: Pick<
-        NonNullable<DeploymentAlert['deployment']>,
-        'id' | 'clusterId' | 'namespace'
-    >;
+export type DeploymentTabWithReadAccessForDeploymentProps = {
+    alertDeployment: AlertDeployment;
 };
 
-function DeploymentDetails({ alertDeployment }: DeploymentDetailsProps): ReactElement {
-    const { hasReadAccess } = usePermissions();
-    const hasReadAccessForNetworkPolicy = hasReadAccess('NetworkPolicy');
-
+function DeploymentTabWithReadAccessForDeployment({
+    alertDeployment,
+}: DeploymentTabWithReadAccessForDeploymentProps): ReactElement {
```

Comment:
These 2 new wrapper component names are very verbose and coupled to the implementation details. Something more concise would be better.

```suggestion
function DeploymentTabWithPermission({
    alertDeployment,
}: DeploymentTabWithPermissionProps): ReactElement {
```